docs_new/features/core/output_types.md

```diff
@@ -0,0 +1,173 @@
+
+output_type = float
+output_type = Dict[int, str]
+output_type = Tuple[Union[str, Dict[int, float]], List[bool], Optional[int]]
+```
+
+### Multiple Choice
+
+You can use your model to do classification by using the `Literal` or `Enum` output types. For instance:
+
+```python
+from typing import Literal
+from enum import Enum
+
+class PizzaOrBurger(Enum):
+    pizza = "pizza
+    burger = "burger
+
+# Equivalent multiple-choice output types
+output_type = Literal["pizza", "burger"]
+output_type = PizzaOrBurger
+```
+
+### Structured Objects
+
+What we call structured object here are roughly all common types that can be turned into a JSON schema. That includes:
+- A Pydantic class
+- A Dataclass
+- A TypedDict
+- A GenSon SchemaBuilder
+- A Callable (the parameters are turned into the keys and the type hinting is used to define the types of the values)
+
+For instance:
+
+```python
+from dataclasses import dataclass
+
+@dataclass
+class Character:
+    name: str
+    age: int
+
+ouput_type = Character
+
+def character(name: str, age: int):
+    return None
+
+output_type = character
+```
+
+On top of those five types that can directly be used with a model, there are also two other json schema formats for which we need to use an Outlines-specific class: JSON schema strings and dicts. As those do not have a distinct type, we must wrap them into an `outlines.types.JsonSchema` object. For instance:
+
+```python
+from outlines.types import JsonSchema
+
+schema_string = '{"type": "object", "properties": {"answer": {"type": "number"}}}'
+output_type = JsonSchema(schema_string)
+
+schema_dict = {
+    "type": "object",
+    "properties": {
+        "answer": {"type": "number"}
+    }
+}
+output_type = JsonSchema(schema_dict)
+```
+
+### Regex Patterns
+
+You can ask your model to generate text that follows a regex pattern. As the latter are expressed as simple raw string literals, we need to use an Outlines-specific class. Thus, the regex string should be be wrapped in an `outlines.types.Regex` object. For instance:
+
+```python
+from outlines.types import Regex
+
+regex = r"[0-9]{3}"
+output_type = Regex(regex)
+```
+
+The `types` module contains a few common regex patterns stored in variable you can import and directly use as output types. There are for instance the patterns for a sentence, an email address and an isbn reference. For instance:
+
+```python
+from outlines.types import sentence
+
+print(type(sentence)) # outlines.types.dsl.Regex
+print(sentence.pattern) # [A-Z].*\s*[.!?]
+```
+
+To help you create complex regex patterns yourself, you can rely on the Outlines [regex DSL]().
+
+### Context-Free Grammars
+
+Outlines allows you to generate text that respects the syntax of a context-free grammar. For that, you need to define a Lark grammar. As the latter is expressed as a string, we need to use an Outlines-specific class. Thus, the CFG string should be be wrapped in an `outlines.types.CFG` object. For instance:
```

Comment:
```suggestion
Outlines allows you to generate text that respects the syntax of a context-free grammar. Context-free grammars are defined using [Lark](https://lark-parser.readthedocs.io/en/latest/index.html), a grammar language. Since grammars are expressed as a string, Large CFG strings should be be wrapped in an `outlines.types.CFG` object. For instance:
```